modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaGetFileMethodCheck.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.check.util.SourceUtil;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaGetFileMethodCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		List<String> importNames = JavaSourceUtil.getImportNames(content);
+
+		if (!importNames.contains(
+				"com.liferay.document.library.kernel.service." +
+					"DLFileEntryLocalServiceUtil") ||
+			!importNames.contains(
+				"com.liferay.document.library.kernel.service." +
+					"DLFileEntryLocalService")) {
+
+			return content;
+		}
+
+		String newContent = content;
+
+		boolean replaced = false;
+
+		Matcher getFileMatcher = _getFilePattern.matcher(newContent);
+
+		while (getFileMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, getFileMatcher.start());
+
+			if (methodCall.contains("DLFileEntryLocalServiceUtil") ||
+				hasClassOrVariableName(
+					"DLFileEntryLocalService", newContent, methodCall)) {
+
+				newContent = _format(newContent, methodCall);
+			}
+
+			replaced = true;
+		}
+
+		if (replaced) {
+			newContent = JavaSourceUtil.addImports(
+				newContent, "com.liferay.portal.kernel.util.FileUtil");
+		}
+
+		return newContent;
+	}
+
+	private String _format(String content, String methodCall) {
+		String[] lines = StringUtil.splitLines(content);
+
+		for (String line : lines) {
+			if (line.contains(methodCall)) {
+				List<String> parameterList = JavaSourceUtil.getParameterList(
+					methodCall);
+
+				if (parameterList.size() != 3) {
+					return content;
+				}
+
+				String newLine = _toAssembleMethod(
+					line, methodCall, parameterList);
+
+				content = StringUtil.replace(content, line, newLine);
+			}
+		}
```

Comment:
This part I do not understand. I'm worried about the performance here because with each match from the regex, we have to iterate __all of the file contents__ each time. Can you tell me why we have to deal the entire file contents line by line like this? Why not just substitute the method call with the new one plus the `File` object declaration?